projects/js-packages/social-logos/tests/tests.js

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env node
+/* eslint-disable no-console, no-process-exit */
+
+// Exit codes:
+//   0: All is well
+//   1: SVG build files are missing
+//   2: The production build isn't clean
+//   3: SVG optimization checks failed
+//   4: Other build files are missing
+
+const { spawnSync } = require( 'child_process' );
+const fs = require( 'fs' );
+const glob = require( 'glob' ); // Add this line to import the 'glob' module
+
+const helperFilesDir = 'tests/helper_files/';
+const svgSrcDir = 'src/svg/';
+const buildDir = 'build/';
+const svgBuildDir = buildDir + 'svg-clean/';
+
+// Start in the right folder.
+const rootDir = __dirname + '/..';
+process.chdir( rootDir );
+
+/**
+ * Checks that each SVG file in the source folder has an equivalent file in the build folder.
+ */
+function checkSVGBuilds() {
+	console.log( 'Verifying all SVGs were built...' );
+	const srcSVGs = glob.sync( '*.svg', { cwd: svgSrcDir } );
+	const builtSVGs = glob.sync( '*.svg', { cwd: svgBuildDir } );
+	for ( const srcSVG of srcSVGs ) {
+		if ( ! builtSVGs.includes( srcSVG ) ) {
+			console.error( `Built SVG file not found: ${ srcSVG }` );
+			process.exit( 1 );
+		}
+	}
+	console.log( 'All SVGs were built.' );
+}
+
+/**
+ * Verifies the SVG optimization is working.
+ */
+function verifySVGOptimization() {
```

Comment:
Ideally this should never fail, but if for some reason the optimization changed defaults or something, it could result in larger (or broken) files. Again, though...an extreme edge case.

Reply:
OTOH, you're testing just one file for an exact match. If the optimization gets better this will also fail, and if it happens to break every file except youtube.svg it won't detect.

That's not to say this is a bad test, just pretty limited.

Reply:
Indeed, quite limited. :^)